api/api/accounts/test_update_users.py

```diff
@@ -0,0 +1,307 @@
+                             'last_name': 'user',
+                             'gender': 'M',
+                             'contact_no': None,
+                             'user_type': 'Student',
+                             'picture_url': '',
+                             'student': OrderedDict([('roll_no', 'changed_roll_no'),
+                                                     ('batch', 2025),
+                                                     ('department', 'CSE'),
+                                                     ('hostel_address', 'H3 Z000'),
+                                                     ('bio', 'zzz')]),
+                             'faculty': None,
+                             'staff': None}
+
+    assert test_user.student.roll_no == 'changed_roll_no'
+
+
+@pytest.mark.django_db
+def testUpdateUserView_facultyUser_changeRequest_updatesSuccessful(client, test_user):
+
+    # given
+    faculty = Faculty.objects.create(user=test_user,              # noqa: F841
+                                     title='Prof',
+                                     department='CSE',
+                                     designation='smthng')
+    client.force_authenticate(test_user)
+
+    # when
+    response = client.put(f'/users/{test_user.id}/', {'email': 'test@user.com',
+                                                      'gender': 'F',
+                                                      'user_type': 'Faculty',
+                                                      'faculty': {
+                                                          'title': 'Prof',
+                                                          'department': 'CSE',
+                                                          'designation': 'smthng_different'}
+                                                      })
+
+    test_user = User.objects.get(id=test_user.id)
+
+    # then
+    assert response.status_code == status.HTTP_200_OK
+    assert response.data == {'email': 'test@user.com',
+                             'first_name': 'test',
+                             'last_name': 'user',
+                             'gender': 'F',
+                             'contact_no': None,
+                             'user_type': 'Faculty',
+                             'picture_url': '',
+                             'student': None,
+                             'faculty': OrderedDict([('title', 'Prof'),
+                                                     ('department', 'CSE'),
+                                                     ('designation', 'smthng_different')]),
+                             'staff': None}
+
+    assert test_user.faculty.designation == 'smthng_different'
+    assert test_user.gender == 'F'
+
+
+@pytest.mark.django_db
+def testUpdateUserView_staffUser_changeRequest_updatesSuccessful(client, test_user):
+
+    # given
+    staff = Staff.objects.create(user=test_user,                # noqa: F841
```

Comment:
Try not to use comments to skip lint checks. They should be added only when there's no way to avoid them.

In this case, you can remove the assignment: `staff = `